verilog/analysis/verilog_linter.cc

```diff
@@ -332,10 +333,51 @@ absl::Status PrintRuleInfo(std::ostream* os,
   return absl::OkStatus();
 }
 
+static void AppendCitation(CustomCitationMap& citations,
+                           const absl::string_view& rule_raw) {
+  const size_t eq_pos = rule_raw.find(':');
+  if (!eq_pos || eq_pos == absl::string_view::npos) return;
+  const size_t rule_id_end = eq_pos;
+  const size_t rule_citation_beg = eq_pos + 1;
+  absl::string_view rule_id = rule_raw.substr(0, rule_id_end);
+  absl::string_view citation =
+      rule_raw.substr(rule_citation_beg, rule_raw.size() - rule_citation_beg);
+
+  citations[rule_id] = absl::StrReplaceAll(citation, {{"\\\n", "\n"}});
+}
+
+CustomCitationMap ParseCitations(absl::string_view text) {
+  CustomCitationMap citations;
+  size_t rule_begin = 0;
+  size_t rule_end = text.find('\n');
+  while (rule_end != std::string::npos) {
+    if (!rule_end || text[rule_end - 1] == '\\') {
+      rule_end = text.find('\n', rule_end + 1);
+      continue;
+    }
+    auto rule_subs = text.substr(rule_begin, rule_end - rule_begin);
+    AppendCitation(citations, rule_subs);
+    rule_begin = rule_end + 1;
+    rule_end = text.find('\n', rule_begin);
+  }
+  return citations;
+}
+
 void GetLintRuleDescriptionsHelpFlag(std::ostream* os,
-                                     absl::string_view flag_value) {
+                                     absl::string_view flag_value,
+                                     const CustomCitationMap& citations) {
   // Set up the map.
   auto rule_map = analysis::GetAllRuleDescriptionsHelpFlag();
+  if (!citations.empty()) {
+    for (const auto& [rule_id, citation] : citations) {
+      auto rule = rule_map.find(rule_id);
+      if (rule != rule_map.end()) {
+        // we can move here, cause process
```

Comment:
I'd just plainly copy it. There is no good reason to save a few instructions and a string-copy at the expense of having a justifying comment clarifying that this will all be good.
(Also moving from a const std::string, how does that work ?)